### ISSUE TITLE
flat:foreign vs. flat:name

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/docs#914]
For languages which decide to use these recommended subtypes, I think the guidelines need clarification.

On the [`flat`](https://universaldependencies.org/u/dep/flat.html) page, the discussion of examples like Ludwig van Beethoven says that any internal compositional structure should be analyzed only if annotating the language in question. So in a German corpus, "van" would attach to "Beethoven" as `case` (it forms a PP), but in an English corpus the whole expression would simply be `flat`.

So how would subtypes apply here? [`flat:foreign`](https://universaldependencies.org/u/dep/flat-foreign.html) flatly (ha) says that foreign names are not included. [`flat:name`](https://universaldependencies.org/u/dep/flat-name.html) gives examples of compositional internal structure in English and Italian, without specifying the language of the corpus. Should it be revised to say that *foreign* syntactic constructions (prepositions, etc.) within a name should be given a `flat:name` analysis rather than a compositional analysis?

**Comment by sylvainkahane** [comment]
`flat:name` should be used between a combination of names: `first name -[flat:name]-> second name`. While _van Beethoven_ is a foreign name and its two parts must be linked by `flat:foreign`. 

Note that this example is a bit complicated, because, even if Beethoven  was German, his name is Dutch and _van_ is not a German word. But it is understandable for any German because the corresponding preposition is _von_: What to do in such case? I think that even in German we should use `flat:foreign`, because we don't want to have _van_ in the German lexicon extracted from a German treebank.

**Comment by sylvainkahane** [comment]
What do you do with middle names in English? For instance, _John F. Kennedy_ is not completely flat, I think. You can call him _John_ or _Kennedy_, but can you call him _John F._ or _F. Kennedy_? Is _F._ a modifier of _John_?

**Comment by amir-zeldes** [comment]
AFAIK the canonical analysis is to connect all parts of the first name via flat, the same for all parts of a middle name, likewise for last, and finally connect all chunks to the first token. So if someone is called:

First: Mary Beth
Middle: K. P.
Last: van Beethoven Smith

It would be:

flat(Mary, Beth)
flat(K., P.)
flat(van, Beethoven)
flat(van, Smith)
flat(Mary, K.)
flat(Mary, van)

I'm reasonably sure it's implemented that way in GUM anyway. If we want "van" to be foreign, then I think it should have a feature expressing that it is a loanword. Syntactically, if we assume that the syntax of "van" is not part of English, then I see no difference in the relationship between "van" and "Bethooven" and "Mary" and "Beth" - both are parts of a multi-token, syntactically headless name.

**Comment by nschneid** [comment]
Oh boy. I would have just treated "Mary Beth K. P. van Beethoven Smith" as one big flat structure (because personal names don't abide by standard properties of headed expressions). I see we have our work cut out for us. :)

**Comment by nschneid** [comment]
The guidelines actually prohibit nesting of `flat` structures: https://universaldependencies.org/u/dep/flat.html

Other dependency types may be nested under a `flat` expression, though. (We clarified this a couple of years ago.)

**Comment by amir-zeldes** [comment]
Mm, sorry, no, I'm sure we've discussed this before and it was clarified that flat may have nested children (other than the first token dominating everything). Here is you saying exactly that last year:

https://github.com/UniversalDependencies/docs/issues/757#issuecomment-791975852

And here is @sylvainkahane agreeing (notwithstanding subtypes):

https://github.com/UniversalDependencies/docs/issues/757#issuecomment-791998046

I'm pretty sure @dan-zeman is also on record saying as much somewhere.

Then linguistically, I think we're all agreed that some multi-word name doesn't really have a head, and that it's not particularly compelling for the first or last name to be the head of the other, but does anyone think a complex first name and a complex last name shouldn't first form their own catena before linking to the other? They are separately deletable ("Ludwig"), interrogable ("Ludwig what?", "Who/which van Beethoven"?) etc., so they fulfill basic phrase tests.

**Comment by nschneid** [comment]
Indeed, they can have nested children, but those children cannot be flat structures per the update discussed here: UniversalDependencies/docs#543

My comment that you linked to was about a nested PP, it seems.

I replied to @sylvainkahane's comment that you linked to with:

> Linguistically speaking I see the logic there—each subunit in the name would be separate—but since the `:name` and `:foreign` subtypes are optional I don't know whether it is practical to have a rule that involves layering of different kinds of `flat` relations.

and then 

>  Even setting aside the subtypes, we have to balance the extra expressivity that would afford a very precise analysis (such as making "van Beethoven" a subunit) vs. a simple and enforceable rule that will prevent errors like a chained `flat` structure when the bouquet structure is correct.

_Originally posted by @nschneid in https://github.com/UniversalDependencies/docs/issues/757#issuecomment-792004224_

I just worry about giving annotators too much leeway here: UD notions of headedness don't provide a complete grammar of names, and saying any part of a name that is not a compositional structure attaches to the first word as `flat` seems simplest to me. I also worry about the complexity of cross-cultural naming conventions, e.g. when a Spanish name has several parts—as an English speaker I may encounter the name and recognize it refers to a person but have no idea of what the different "slots" are in the pattern and which parts are optional etc. That feels like a task for detailed NER, not for sentence syntax.

**Comment by amir-zeldes** [comment]
> Indeed, they can have nested children, but those children cannot be flat structures per the update discussed here

I just read #543 and do not see the sentence saying the flat children cannot have internal flat dependencies. @sylvainkahane says in that issue:

> AB is flat if you cannot decide which of A and B is the head. But A and B can be complex phrases.

Where does it say that internally, A and B cannot contain flat? I think for the case of multi-part names there is a strong motivation to allow this. Prohibiting it categorically would have strange consequences. If you think I resemble Dustin Hoffman (which I don't :) and you want to nickname me:

  * Amir "Dustin Hoffman" Zeldes

Do you think it's right to split apart Dustin Hoffman, just because it is a flat dependent of "Amir"? Note that this would also create trouble for the punctuation attachment guideline (the quotation marks could not be dominated by the internal head of the surrounded nickname without non-projectivity), and speaking of NER, if you wanted to annotate the large unit as me, and Dustin Hoffman as a mention of Dustin Hoffman, then that would be a non-catena mention with no well defined head. 

Add to that the fact that these are really separate linguistic phrases by classic linguistic tests, and I don't really understand why you would want to prohibit this. The fact that you, or I, or some other speaker might not know how to parse a Spanish name correctly does not mean that it does not have a correct division into first and last names, and a guideline on what to do with first, middle and last names does not seem so exotic or hard to convey to annotators. It's syntactic and not a task for NER for the simple fact that each part is separately omittable and interrogable, and in fact it's not a task for NER, since NER usually has a monolithic PER span, with no division of first and last name.

**Comment by nschneid** [comment]
I think of "flat" as being agnostic as to structure, and saying there is no single head within the realm of the general syntax of the language. I.e. it would not really be saying that "Dustin Hoffman" is NOT a unit, it's just abstaining from analyzing the substructure.

I guess it's hard to follow the thread in #543 because it mixes a discussion of `fixed`. @martinpopel at https://github.com/UniversalDependencies/docs/issues/543#issuecomment-381312960 gave the caveat of non-`flat` dependents. @dan-zeman confirmed that he was OK with my revision of the guidelines.

I'm certainly open to revisiting this in the group if there would be enthusiasm for that.

**Comment by amir-zeldes** [comment]
I read @martinpopel 's comment as requesting an example where B has non-flat dependents, not as prohibiting any and all cases where B might have flat dependents. Sorry I missed this implication in your edit at the time, I had simply understood that A and B can be complex, which makes sense to me.

For me, saying that something is flat does mean that any incomplete subset of its tokens is not a unit. Otherwise, the thing that is a unit is a "B", and the flat relation reduces to existing between A and B. If B itself, while being a unit, does not have a clear head, then I think B is internally flat. That's my understanding of "A and B can be complex" (i.e. the same rules apply to them as apply to any sequence of tokens in UD, with corresponding analyses).

Does anyone feel that it is linguistically incorrect to group "Dustin Hoffman" in the example above as a dependency chain?

**Comment by jnivre** [comment]
When we introduced ”flat” for version 2, my understanding was always that the things related by this relation could be complex phrases.

Joakim

**Comment by nschneid** [comment]
@jnivre So you would allow structures of the form X -flat-> Y -flat-> Z? And interpret it as nested nonheaded constituents, [X [Y Z]]?

**Comment by jnivre** [comment]
Yes. Just like we can do with “compound” (but not with “fixed”).

Skickat från min iPhone

8 dec. 2022 kl. 23:06 skrev Nathan Schneider ***@***.***>:

﻿

@jnivre<https://github.com/jnivre> So you would allow structures of the form X -flat-> Y -flat-> Z? And interpret it as nested nonheaded constituents, [X [Y Z]]?

**Comment by nschneid** [comment]
Interesting. @dan-zeman are you OK with making "`flat` dependencies must go from left to right" the only formal constraint on flat structures?

**Comment by dan-zeman** [comment]
> are you OK with making "`flat` dependencies must go from left to right" the only formal constraint on flat structures?

If I'm not mistaken, this is the only constraint currently enforced by the UD validator.

Unlike Joakim, I can't remember whether I foresaw `flat` dependents as potentially complex structures when we were introducing them (`flat` was replacing two UD v1 relations, `name` and `foreign`, and while the definition of `name` may have not banned complex dependents, I don't think I ever encountered an example). But the main distinguishing property of `flat` is that two or more constituents belong together and none of them can be considered the head (plus they are not anything more specific, like coordination). Not willing to pick a head among N constituents is not the same thing as not allowing internal structure of the participating constituents, so nested `flat` structures look technically plausible to me. I'm not sure though that I like the analysis of complex names proposed by @amir-zeldes above. (BTW nested flat structures have the same drawback as nested coordination: Some types cannot be distinguished by annotation. For example, _((Александр Сергеевич) Пушкин)_ cannot be distinguished from _(Александр Сергеевич Пушкин)_. In both cases, there will be just `flat(Александр, Сергеевич); flat(Александр, Пушкин)`.)

**Comment by nschneid** [comment]
If flat structures are allowed to nest within larger flat structures, then we have to decide the boundaries between `flat:foreign` and `flat:name` in languages that use the subtypes. I.e., can a `flat:foreign` structure be nested within a `flat:name` structure? How should a part of a name with foreign structure (like "van Beethoven" in English) be treated? Does it rest on the assumption that we can recognize some foreign words as being function words rather than nouns (maybe relying on capitalization)?

On https://universaldependencies.org/u/dep/flat-name.html, it says that *organization* names can have nested PP structure. I don't understand the difference between examples 3 and 6 (but maybe these are intended for Italian annotation).

**Comment by jnivre** [comment]
I would not use the subtype “foreign” at all for names like van Beethoven. I think it should be reserved for genuine cases of code-switching.

Joakim

**Comment by dan-zeman** [comment]
The original/official position was that the universal guidelines do not give definitions of relation subtypes, although we are slowly departing from this strict stance. So some widely used subtypes now have a globally accessible documentation page, which may (but does not have to) be overriden by a language-specific page. The primary motivation here was technical: to not discard a language as invalid just because it failed to provide its own version of documentation of something that is used in many languages. Now of course it is also desirable to make sure that all languages use the subtype consistently, but my feeling is that finding the precise cross-linguistic borderline between two subtypes of the same main type is a less burning issue, as long as the main type is used correctly.

In fact, many of the globally accessible doc pages of subtypes were created by me, and I simply picked one of the already existing language-specific pages and copied it to `u-dep`. That's why the current global doc of `flat:name` has Italian examples and `flat:foreign` has a Czech example.

**Comment by nschneid** [comment]
Do you think a disclaimer is warranted on such pages? ("This subtype has a universal documentation page because it is used by several languages, but the documentation below is not intended as a strict standard: it has not been reviewed by the UD Core Group, and languages using the relation may vary in their interpretations.")

**Comment by amir-zeldes** [comment]
> If I'm not mistaken, this is the only constraint currently enforced by the UD validator.

I can confirm this is so (and I'm fine with it this way!)

> For example, ((Александр Сергеевич) Пушкин) cannot be distinguished from (Александр Сергеевич Пушкин)

Yes, this is a fundamental limitation of the 'first dibs' attachment left-to-right, which affects conj etc. in the same way. I don't view it as an argument against nevertheless attempting to represent as much as possuble using basic dependencies (just as we do with nested coordination)

**Comment by dan-zeman** [comment]
> Do you think a disclaimer is warranted on such pages? ("This subtype has a universal documentation page because it is used by several languages, but the documentation below is not intended as a strict standard: it has not been reviewed by the UD Core Group, and languages using the relation may vary in their interpretations.")

Not sure. While I said I don't see it as a burning issue, I still think that improving the text is better than telling people plainly they can do with it whatever they want. Second-class status is somewhat reflected already by not including subtypes in the [relation table](https://universaldependencies.org/u/dep/index.html). A disclaimer about not being reviewed by the Core Group would apply to all the language-specific pages, too...

**Comment by nschneid** [comment]
I'm all for improving text where possible but my concern is that the "universal" guidelines are not a coherent set of recommendations, since there are some "important" subtypes (like `acl:relcl` or `nsubj:outer`) and some other idiosyncratic subtypes whose boundaries with other subtypes are not clear.

If we are going to have a coherent story for `flat`, it should either be "subtypes are optional and languages may interpret them differently, use them at your own risk" or "should you choose to use subtypes, here are the criteria for deciding which one is suitable". For a subtype that has only been used in a few treebanks, perhaps the former strategy (not rushing into a premature definition) is better. I just find it confusing to have half-baked "universal" guidelines pages without a warning to that effect.
